api/tests/unit/organisations/test_unit_organisations_views.py

```diff
@@ -657,6 +659,44 @@ def test_when_subscription_is_set_to_non_renewing_then_cancellation_date_set_and
     mocked_cancel_chargebee_subscription.assert_not_called()
 
 
+@pytest.mark.freeze_time("2023-01-19T09:09:47.325132+00:00")
+@mock.patch("organisations.models.cancel_chargebee_subscription")
+def test_when_subscription_is_set_to_non_renewing_then_cancellation_date_set_and_current_term_end_is_missing(
+    mocked_cancel_chargebee_subscription: MagicMock,
+    subscription: Subscription,
+    staff_user: FFAdminUser,
+    staff_client: APIClient,
+    settings: SettingsWrapper,
+) -> None:
+    # Given
+    subscription.subscription_id = "subscription-id"
+    subscription.save()
+
+    data = {
+        "content": {
+            "subscription": {
+                "status": "non_renewing",
+                "id": subscription.subscription_id,
+                # Note the missing current_term_end field.
+            },
+            "customer": {"email": staff_user.email},
+        }
+    }
+    url = reverse("api-v1:chargebee-webhook")
+
+    settings.ORG_SUBSCRIPTION_CANCELLED_ALERT_RECIPIENT_LIST = ["foo@bar.com"]
+
+    # When
+    staff_client.post(url, data=json.dumps(data), content_type="application/json")
+
+    # Then
+    subscription.refresh_from_db()
+    assert subscription.cancellation_date is None
```

Comment:
This feels like the opposite of what we want to test?

Reply:
It gets set to `None` because if a canceled subscription isn't slated in the future it reverts to being a free plan, so the cancelation date is set to `None`.

Reply:
I've added a comment to the assertion to clarify why it is set to `None`.